lab-kyle/route/user-routes.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+let User = require('../model/user')
+let authMiddleware = require('../lib/auth')
+let jsonParser = require('body-parser').json()
+
+module.exports = (router) => {
+
+  router.post('/users', jsonParser, (req, res) => {
+    let user = new User(req.body)
+    user.hashPassword(user.password)
+      .then(user => user.save())
+      .then(user => res.json(user.username))
+      .catch(() => res.status(400).send('invalid body'))
+  })
+
+  router.get('/users/:id', authMiddleware, (req, res) => {
+    User.findById(req.params.id)
+      .then(user => {
+        delete req.user.password
```

Comment:
There's an issue here in that there's no user object on the request object, which is why this fails.

To finish this up you need to add the user to the request object inside of the auth middleware.